cli-plugins/socket/socket_test.go

```diff
@@ -178,24 +178,39 @@ func TestConnectAndWait(t *testing.T) {
 	// TODO: this test cannot be executed with `t.Parallel()`, due to
 	// relying on goroutine numbers to ensure correct behaviour
 	t.Run("connect goroutine exits after EOF", func(t *testing.T) {
+		runtime.LockOSThread()
+		defer runtime.LockOSThread()
```

Comment:
I'm guessing this was meant to be

```suggestion
		defer runtime.UnlockOSThread()
```